xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm

```diff
@@ -167,11 +167,17 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
       #if ($hasAdmin || ($hasEdit && $isUserComment))
         ## If a remote URL is provided, content will be loaded into .modal-content because of bootstrap.
         ## By providing an anchor this behavior is stoped, without altering the URL functionality.
+        #set ($deleteQueryString = $escapetool.url({
+          'form_token': $services.csrf.token,
+          'classname': $xCommentClass,
+          'classid': $comment.number,
+          'xredirect': $doc.getURL('view')
+        }))
+        #set ($deleteURL = $doc.getURL('objectremove', "${deleteQueryString}#xwikicomment_${comment.number}"))
```

Comment:
You can use  `XWiki#getURL(String fullname, String action, String querystring, String anchor)` instead of appending the anchor to the queryString.